example/run.sh

```diff
@@ -210,15 +208,74 @@ echo ""
 echo "Calculate metrics after inference..."
 echo ""
 python $root_dir/calculate_baseline_metrics.py \
-    --label_file $out_dir/test_data.h5 --task regression \
-    --test_file $out_dir/inference.pretrained_latest/no_label_HSC.5M.output.pretrained.track.bw \
+    --label_file $data_dir/HSC.80M.chr123.10mb.coverage.bw \
+    --task regression \
+    --test_file $out_dir/inference.pretrained_latest/test_data_HSC.5M.output.pretrained.track.bw \
     --intervals $out_dir/example.holdout_intervals.bed \
     --sizes $ref_dir/hg19.auto.sizes \
-    --sep_peaks
+    --sep_peaks --peak_file $out_dir/HSC.80M.chr123.10mb.peaks.bed.bw
 
 python $root_dir/calculate_baseline_metrics.py \
-    --label_file $out_dir/test_data.h5 --task classification \
-    --test_file $out_dir/inference.pretrained_latest/no_label_HSC.5M.output.pretrained.peaks.bw \
+    --label_file $out_dir/HSC.80M.chr123.10mb.peaks.bed.bw --task classification \
+    --test_file $out_dir/inference.pretrained_latest/test_data_HSC.5M.output.pretrained.peaks.bw \
     --intervals $out_dir/example.holdout_intervals.bed \
     --sizes $ref_dir/hg19.auto.sizes \
-    --thresholds 0.5
+    --auc --thresholds 0.5
+    
+########
+
+echo ""
+echo "Test the ability of the model to take additional inputs"
+echo ""
+
+# Encode data with CTCF motif layer
+python $root_dir/bw2h5.py \
+    --noisybw $data_dir/HSC.5M.chr123.10mb.coverage.bw \
+    --intervals $out_dir/example.training_intervals.bed \
+    --batch_size 4 \
+    --prefix $out_dir/train_data_layers \
+    --cleanbw $data_dir/HSC.80M.chr123.10mb.coverage.bw \
+    --cleanpeakbw $out_dir/HSC.80M.chr123.10mb.peaks.bed.bw \
+    --layersbw ctcf_for:$data_dir/ctcf_motifs_for.sorted.bg.bw \
+    --nonzero
+# Validation data
+python $root_dir/bw2h5.py \
+    --noisybw $data_dir/HSC.5M.chr123.10mb.coverage.bw \
+    --intervals $out_dir/example.val_intervals.bed \
+    --batch_size 64 \
+    --prefix $out_dir/val_data_layers \
+    --cleanbw $data_dir/HSC.80M.chr123.10mb.coverage.bw \
+    --cleanpeakbw $out_dir/HSC.80M.chr123.10mb.peaks.bed.bw \
+    --layersbw ctcf_for:$data_dir/ctcf_motifs_for.sorted.bg.bw
+# Test data
+python $root_dir/bw2h5.py \
+    --noisybw $data_dir/HSC.5M.chr123.10mb.coverage.bw \
+    --layersbw ctcf_for:$data_dir/ctcf_motifs_for.sorted.bg.bw \
+    --intervals $out_dir/example.holdout_intervals.bed \
+    --batch_size 64 \
+    --prefix $out_dir/test_data_layers \
+    --nolabel
+
+#Train and validate model..."
+python $root_dir/main.py --train \
+    --train_files $out_dir/train_data_layers.h5 \
+    --val_files $out_dir/val_data_layers.h5 \
+    --out_home $out_dir --label HSC.5M.model.layers \
+    --layers ctcf_for \
+    --in_channels 2 \
```

Comment:
@avantikalal can the channels be automatically determined by how many layers need to be used?

Reply:
@tijyojwad I suppose so. But I thought that the number of input channels should be included in the model parameters file as it's an important part of the model structure.

Reply:
Fair enough, I'm okay with it being there. Also adds another layer of sanity.